### PR TITLE
Release v1.31.0

### PR DESCRIPTION
**Pull request by rashmigottipati**
**Description of the change:**
Prep for the v1.31.0 release

**Motivation for the change:**
Release v1.31.0 of the operator-sdk

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

File: testdata/ansible/memcached-operator/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ endif
 
 # Set the Operator SDK version to use. By default, what is installed on the system is used.
 # This is useful for CI or a project to utilize a specific version of the operator-sdk toolkit.
-OPERATOR_SDK_VERSION ?= v1.30.0
+OPERATOR_SDK_VERSION ?= v1.28.0
```
Comment (rashmigottipati) [inline testdata/ansible/memcached-operator/Makefile:51]: Not sure why this is not being updated to `1.31.0` automatically by the release script.

Comment (oceanc80) [reply testdata/ansible/memcached-operator/Makefile:51]: Yeah, odd that it's updating it backwards...

Comment (varshaprasad96) [reply testdata/ansible/memcached-operator/Makefile:51]: @rashmigottipati this version is taken from the binary that is available locally, (ie) here: https://github.com/operator-framework/operator-sdk/blob/c6d1bf979e86dc0c055060e3872f5e7bbd629f3f/internal/plugins/manifests/v2/init.go#L72

Which is again obtained from here: https://github.com/operator-framework/operator-sdk/blob/c6d1bf979e86dc0c055060e3872f5e7bbd629f3f/Makefile#L9

We can leave this to 1.30.0 (which is the latest available one), till we release 1.31. @rashmigottipati you could fetch all the latest available tags (could run `git fetch --all`), and generate the testate again. That should bump it back to 1.30.

File: changelog/generated/v1.31.0.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+## v1.31.0
+
+### Changes
+
+- (ansible): Update the `quay.io/operator-framework/ansible-operator` base image to now use Ansible 2.15.0.  Removes pre-installed Python modules that are not required to build and run the `testdata/ansible/memcached-operator` sample. Updates Python to 3.9 to work with Ansible 2.15.0. ([#6483](https://github.com/operator-framework/operator-sdk/pull/6483))
+- (ansible/v1): Updates the `ansible/v1` plugin scaffolding to: - Remove the `community.kubernetes` collection - Update the `operator_sdk.util` collection from `v0.4.0` --> `v0.5.0` - Update the `molecule.yaml` files to work with `molecule v5.1.0`. ([#6483](https://github.com/operator-framework/operator-sdk/pull/6483))
+- (helm): Use informer to list helm secrets to improve performance. ([#6354](https://github.com/operator-framework/operator-sdk/pull/6354))
+- Updated Reference from " to_be_owned.namespace" to " to_be_owned.metadata.namespace " at line 186.
```
Comment (varshaprasad96) [inline changelog/generated/v1.31.0.md:8]: This is a part of helm? Maybe we missing a reference PR here, or it is a part of previous one?

File: testdata/ansible/memcached-operator/Makefile (hunk 1 above)
Comment (rashmigottipati) [reply testdata/ansible/memcached-operator/Makefile]: Thanks @varshaprasad96, just fixed it :) It should be updated now

File: changelog/generated/v1.31.0.md (hunk 2 above)
Comment (rashmigottipati) [reply changelog/generated/v1.31.0.md]: It is missing a reference PR, I just added the link to the PR so it should be updated now.

File: changelog/generated/v1.31.0.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+- (helm): Use informer to list helm secrets to improve performance. ([#6354](https://github.com/operator-framework/operator-sdk/pull/6354))
+- Updated Reference from "to_be_owned.namespace" to "to_be_owned.metadata.namespace " in the ansible reference documentation. ([#6409](https://github.com/operator-framework/operator-sdk/pull/6409))
+
+### Removals
+
+- (ansible): Remove the `quay.io/operator-framework/ansible-operator-2.11-preview` image. This image will no longer be built after Operator-SDK v1.30.0. ([#6483](https://github.com/operator-framework/operator-sdk/pull/6483))
+
+### Bug Fixes
+
+- Fix a bug where `olm install` command is failed fo "no-match" error.
```
Comment (varshaprasad96) [inline changelog/generated/v1.31.0.md:16]: nit
```suggestion
- Fix a bug where `olm install` command is failed for "no-match" error.
```

File: changelog/generated/v1.31.0.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+The output in this case is something like:
+```$ operator-sdk olm install --verbose ... FATA[0001] Failed to install OLM version "latest": failed to create CRDs and resources: no matches for kind "OLMConfig" in version "operators.coreos.com/v1"```. Now, in this case, operator-sdk tries to create the resource again, until it succeed (or until the timeout exceeded). ([#6490](https://github.com/operator-framework/operator-sdk/pull/6490))
+- Fix a bug where `run bundle` command does not copy all FBC manifests into the new catalog image if custom index-image is passed. ([#6512](https://github.com/operator-framework/operator-sdk/pull/6512))
```
Comment (varshaprasad96) [inline changelog/generated/v1.31.0.md:19]: nit
```suggestion
```$ operator-sdk olm install --verbose ... FATA[0001] Failed to install OLM version "latest": failed to create CRDs and resources: no matches for kind "OLMConfig" in version "operators.coreos.com/v1"```. Now, in this case, operator-sdk tries to create the resource again, until it succeeds (or until the timeout exceeded). ([#6490](https://github.com/operator-framework/operator-sdk/pull/6490))
- Fix a bug where `run bundle` command does not copy all FBC manifests into the new catalog image if custom index-image is passed. ([#6512](https://github.com/operator-framework/operator-sdk/pull/6512))
```

File: changelog/generated/v1.31.0.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+- markUnsafe now correctly marks as unsafe the spec extra variable. ([#6376](https://github.com/operator-framework/operator-sdk/pull/6376))
```
Comment (varshaprasad96) [inline changelog/generated/v1.31.0.md:20]: nit:
```suggestion
- `markUnsafe` now correctly marks as unsafe the all variables passed in `spec`. ([#6376](https://github.com/operator-framework/operator-sdk/pull/6376))
```

**Pull request merged**